resources/windows/wix/filter.xslt.erb

```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<xsl:stylesheet version="1.0"
+ xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
+ xmlns:wix="http://schemas.microsoft.com/wix/2006/wi">
+  <!-- https://ahordijk.wordpress.com/2013/03/26/automatically-add-references-and-content-to-the-wix-installer/ -->
+  <!-- http://www.chriskonces.com/using-xslt-with-heat-exe-wix-to-create-windows-service-installs/ -->
+  <xsl:output method="xml" indent="yes" />
+  <!--<xsl:strip-space elements="*"/>-->
+  <xsl:template match="@*|node()">
+    <xsl:copy>
+      <xsl:apply-templates select="@*|node()"/>
+    </xsl:copy>
+  </xsl:template>
+  <!-- Filter out all Service File Executables from the Harvest (heat) run as these are specified in the transformed service.components.wxs file -->
+  <!-- The list of component service files are collected in an array so that a set of unique names can be extracted -->
+  <!-- we have to substitue the SourceDir\\etc. etc. for $(var.ProjectSourcePath) in order to facilitate heat -->
+  <!--   being run in two different places, specifically it no longer runs from SourceDir so we needed to pass -->
+  <!--   in ProjectSourcePath to the heat runs. Thus when this filter does its work the source attribute will -->
+  <!--   look like $(var.ProjectSourcePath)/puppet/bin/rubyw.exe for example, not  -->
+  <!--   SourceDir/Program64FilesFolder/Puppet/puppet/bin/ruby.exe -->
+  <%- service_files = Array.new -%>
```

Comment:
So, this is part of how we needed to implement WiX with vanagon. Normally, this functionality (filtering out files from a heat run) would exist in vanagon. However, the implementation we use in vanagon will *only* filter files listed as services in the component/project configuration (see [here](https://github.com/puppetlabs/puppet-agent/blob/master/configs/components/puppet.rb#L48) for an example of something listed as a service file by vanagon)

All the following lines do is add things listed as service files to an XML filter that we feed to heat.exe when it's autogenerating all the WiX for the project.

As I said before, normally this functionality is located in vanagon, but since we are adding the new puppetres.dll file to the filter (and we cant list it as a service because that breaks things) we needed to copy over this functionality to the agent so we could add things to the filter.

We would like to go back to using the file in vanagon and remove this in the future, but https://tickets.puppetlabs.com/browse/VANAGON-46 needs to be resolved before we can do so